metadata_service/proxy/atlas_proxy.py

```diff
@@ -652,6 +689,66 @@ def _parse_date(self, date: int) -> Optional[int]:
         except Exception:
             return None
 
+    @staticmethod
+    def _extract_entities(collections: EntityCollection) -> List[Entity]:
+        """
+        Helper method for flattening all collections from {collections}
+
+        :return: list of all entities
+        """
+        entities: List[Entity] = []
+
+        for collection in collections:
+            entities.extend(collection.entities)
+        return entities
+
+    def _get_readers(self, qualified_name: str, top: Optional[int] = None) -> List[Reader]:
+        params = {
+            'typeName': self.READER_TYPE,
+            'offset': '0',
+            'limit': top or 15,
+            'excludeDeletedEntities': True,
+            'entityFilters': {
+                'condition': 'AND',
+                'criterion': [
+                    {
+                        'attributeName': self.QN_KEY,
+                        'operator': 'STARTSWITH',
+                        'attributeValue': qualified_name.split('@')[0]
+                    },
+                    {
+                        'attributeName': self.READER_COUNT_KEY,
+                        'operator': 'gt',
+                        'attributeValue': '0'
+                    }
+                ]
+            },
+            'attributes': [self.READER_COUNT_KEY, self.QN_KEY],
+            'sortBy': self.READER_COUNT_KEY,
+            'sortOrder': 'DESCENDING'
+        }
+
+        search_results = self._driver.search_basic.create(data=params, ignoreRelationships=False)
+
+        readers = []
+
+        for record in search_results.entities:
```

Comment:
I think the user ID will also be available in `record` here. under related attributes, as the qualified name of User. you don't need to make extra call based on reader IDs, just for the user id. Can you please check ?

Reply:
I think basic search doesn't return relationship attributes. I checked it and the attribute `relationshipAttributes` is always None